File: src/main/java/com/mellivorines/generator/exception/BizException.java
package com.mellivorines.generator.exception;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

public class BizException extends RuntimeException {

    private static final long serialVersionUID = 1L;
    private int code;
    private String msg;

    public BizException(String msg) {
        super(msg);
        this.code = ErrorCode.INTERNAL_SERVER_ERROR.getCode();
        this.msg = msg;
    }

    public BizException(ErrorCode errorCode) {
        super(errorCode.getMsg());
        this.code = errorCode.getCode();
        this.msg = errorCode.getMsg();
    }

    public BizException(String msg, Throwable e) {
        super(msg, e);
        this.code = ErrorCode.INTERNAL_SERVER_ERROR.getCode();
        this.msg = msg;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (!(o instanceof BizException)) return false;

        BizException that = (BizException) o;

        return new EqualsBuilder().append(getCode(), that.getCode()).append(getMsg(), that.getMsg()).isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(17, 37).append(getCode()).append(getMsg()).toHashCode();
    }
}

File: src/main/kotlin/com/mellivorines/generator/exception/BizException.kt
package com.mellivorines.generator.exception

class BizException : RuntimeException {
    var code: Int
    var msg: String

    constructor(msg: String) : super(msg) {
        this.code = ErrorCode.INTERNAL_SERVER_ERROR.code
        this.msg = msg
    }

    constructor(errorCode: ErrorCode) : super(errorCode.msg) {
        this.code = errorCode.code
        msg = errorCode.msg
    }

    constructor(msg: String, e: Throwable?) : super(msg, e) {
        this.code = ErrorCode.INTERNAL_SERVER_ERROR.code
        this.msg = msg
    }


    companion object {
        private const val serialVersionUID = 1L
    }
}